pkg/stub/handler.go

```diff
@@ -432,32 +435,22 @@ func (h *Handler) buildSkipFilters(opcfg *v1alpha1.SamplesResource) {
 	}
 }
 
-func (h *Handler) processError(opcfg *v1alpha1.SamplesResource, ctype v1alpha1.SamplesResourceConditionType, err error, msg string, args ...interface{}) error {
+func (h *Handler) processError(opcfg *v1alpha1.SamplesResource, ctype v1alpha1.SamplesResourceConditionType, cstatus corev1.ConditionStatus, err error, msg string, args ...interface{}) error {
 	log := ""
 	if args == nil {
 		log = fmt.Sprintf(msg, err)
 	} else {
 		log = fmt.Sprintf(msg, err, args)
 	}
 	logrus.Println(log)
-	var status *v1alpha1.SamplesResourceCondition
-	switch ctype {
-	case v1alpha1.SecretUpdateFailed:
-		status = opcfg.Condition(v1alpha1.ImportCredentialsExist)
-	case v1alpha1.SamplesUpdateFailed:
-		status = opcfg.Condition(v1alpha1.SamplesExist)
-	case v1alpha1.InstallTypeChangeDetected, v1alpha1.ArchitectureChangeDetected:
-		status = opcfg.Condition(v1alpha1.ConfigurationValid)
-	default:
-		return err
-	}
+	status := opcfg.Condition(ctype)
```

Comment:
i think there is a bug here.

If opcfg does not contain the condition "ctype", Condition() manufactures one and returns it.

In the case that one was manufactured, there's no guarantee you ultimately proceed to call sdkwrapper.Update() to actually update the resource w/ the new status that got manufactured.

Reply:
I changed `Condition(ctype)` to return a condition with the Status field set to an empty string to account for this.

Callers should be passing in one of the Condition constants.  None of those are empty string.  They would have to pass in `ConditionStatus("")`  to cause a case where one that was manufactured did not lead to a resource update.

Do you want me to account for that case?

Reply:
i guess it's sufficient.